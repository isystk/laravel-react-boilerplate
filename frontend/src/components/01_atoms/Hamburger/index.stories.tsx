import { Meta, Story } from '@storybook/react'
import React from 'react'
import Hamburger, { HamburgerProps } from './index'

export default {
  title: '01_atoms/Hamburger',
  component: Hamburger,
} as Meta

const Template: Story = ({ isMenuOpen }) => {
  const props: HamburgerProps = {
    isMenuOpen,
    setMenuOpen: (isMenuOpen) => console.log(isMenuOpen),
  }
  return <Hamburger {...props} />
}

export const Close = Template.bind()
Close.storyName = 'Close'
Close.args = { isMenuOpen: false }
export const Open = Template.bind()
Open.storyName = 'Open'
Open.args = { isMenuOpen: true }
