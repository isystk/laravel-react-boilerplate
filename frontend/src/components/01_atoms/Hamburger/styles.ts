import { css, keyframes } from '@emotion/css'

export const hamburger = css`
  zoom: 80%;
`
