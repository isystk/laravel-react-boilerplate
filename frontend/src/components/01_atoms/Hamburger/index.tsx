import React, { FC } from 'react'
import { ContainerProps } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** HamburgerProps Props */
export type HamburgerProps = {
  isMenuOpen: boolean
  setMenuOpen: (isMenuOpen: boolean) => void
}
/** Presenter Props */
export type PresenterProps = HamburgerProps

/** Presenter Component */
const HamburgerPresenter: FC<PresenterProps> = ({
  isMenuOpen,
  setMenuOpen,
}) => (
  <>
    <button
      onClick={() => setMenuOpen(!isMenuOpen)}
      className={`space-y-2 ${styles.hamburger}`}
    >
      <div
        className={
          isMenuOpen
            ? 'w-6 h-0.5 bg-black translate-y-2.5 rotate-45'
            : 'w-6 h-0.5 bg-black'
        }
      />
      <div
        className={
          isMenuOpen ? 'w-6 h-0.5 bg-black -rotate-45' : 'w-6 h-0.5 bg-black'
        }
      />
      <div className={isMenuOpen ? 'opacity-0' : 'w-6 h-0.5 bg-black'} />
    </button>
  </>
)

/** Container Component */
const HamburgerContainer: React.FC<
  ContainerProps<HamburgerProps, PresenterProps>
> = ({ presenter, ...props }) => {
  return presenter({
    ...props,
  })
}

export default connect<HamburgerProps, PresenterProps>(
  'Hamburger',
  HamburgerPresenter,
  HamburgerContainer
)
