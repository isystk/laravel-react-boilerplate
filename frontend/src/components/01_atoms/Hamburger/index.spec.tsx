import React from 'react'
import renderer from 'react-test-renderer'
import Hamburger, { HamburgerProps } from './index'

describe('Hamburger', () => {
  it('Match Snapshot', () => {
    const props: HamburgerProps = {
      isMenuOpen: false,
      setMenuOpen: (isMenuOpen) => console.log(isMenuOpen),
    }
    const component = renderer.create(<Hamburger {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
