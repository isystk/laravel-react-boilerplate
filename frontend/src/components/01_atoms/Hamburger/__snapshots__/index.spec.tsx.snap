// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Match Snapshot 1`] = `
<button
  className="space-y-2 css-jgxxba"
  onClick={[Function]}
>
  <div
    className="w-6 h-0.5 bg-black"
  />
  <div
    className="w-6 h-0.5 bg-black"
  />
  <div
    className="w-6 h-0.5 bg-black"
  />
</button>
`;
